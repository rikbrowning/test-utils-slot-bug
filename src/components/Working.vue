<template>
  <FakeComponent>
    <TestComponent :msg="msg" slot="test">
      <div>
        default slot of test component
      </div>
    </TestComponent>
    <div>
      default slot of fake component
    </div>
  </FakeComponent>
</template>

<script>
import TestComponent from "./TestComponent.vue";
import FakeComponent from "./FakeComponent.vue";
export default {
  name: "Working",
  props: {
    msg: String,
  },
  components: {
    FakeComponent,
    TestComponent,
  },
};
</script>
