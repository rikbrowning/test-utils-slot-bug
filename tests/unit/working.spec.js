import { shallowMount } from "@vue/test-utils";
import Working from "@/components/Working.vue";

describe("Working.vue", () => {
  it("renders props.msg when passed", () => {
    const msg = "new message";
    const wrapper = shallowMount(Working, {
      propsData: { msg },
    });
    expect(wrapper).toMatchSnapshot();
  });
});
