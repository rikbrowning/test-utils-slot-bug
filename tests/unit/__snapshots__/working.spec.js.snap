// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Working.vue renders props.msg when passed 1`] = `
<fakecomponent-stub>
  <testcomponent-stub msg="new message">
    <div>
      default slot of test component
    </div>
  </testcomponent-stub>
  <div>
    default slot of fake component
  </div>
</fakecomponent-stub>
`;
